model/riscv_insts_base.sail

```diff
@@ -317,7 +317,61 @@ function is_aligned(vaddr : xlenbits, width : word_width) -> bool =
 function check_misaligned(vaddr : xlenbits, width : word_width) -> bool =
   not(plat_enable_misaligned_access()) & not(is_aligned(vaddr, width))
 
+function transform_VA (effective_address : xlenbits,pmlen : {7, 16}) -> xlenbits = {
+  let XLEN = sizeof(xlen);
+  let i = XLEN -(pmlen +1);
+  let effective_bit = effective_address[i..i];
+  let masked_bits = replicate_bits(effective_bit, pmlen);
+  let unmasked_bits = effective_address[i..0];
+  masked_bits @ unmasked_bits
+}
+
+function transform_PA (effective_address : xlenbits,pmlen : {7, 16}) -> xlenbits = {
+  let XLEN = sizeof(xlen);
+  let i = XLEN -(pmlen +1);
+  let masked_bits = replicate_bits(0b0, pmlen);
+  let unmasked_bits = effective_address[i..0];
+  masked_bits @ unmasked_bits
+}
```

Comment:
This function can be similarly simplified:

```
function transform_PA(effective_address : xlenbits, pmlen : {7, 16}) -> xlenbits =
  zero_extend(effective_address[sizeof(xlen) - pmlen - 1 .. 0])
```

Reply:
Done